baselines/common/mpi_adam_optimizer.py

```diff
@@ -17,30 +17,27 @@ def compute_gradients(self, loss, var_list, **kwargs):
         num_tasks = self.comm.Get_size()
         buf = np.zeros(sum(sizes), np.float32)
 
-        sess = tf.get_default_session()
-        assert sess is not None
         countholder = [0] # Counts how many times _collect_grads has been called
         stat = tf.reduce_sum(grads_and_vars[0][1]) # sum of first variable
-        def _collect_grads(flat_grad):
+        def _collect_grads(flat_grad, stat):
             self.comm.Allreduce(flat_grad, buf, op=MPI.SUM)
             np.divide(buf, float(num_tasks), out=buf)
             if countholder[0] % 100 == 0:
-                check_synced(sess, self.comm, stat)
+                check_synced(self.comm, stat)
             countholder[0] += 1
             return buf
 
-        avg_flat_grad = tf.py_func(_collect_grads, [flat_grad], tf.float32)
+        avg_flat_grad = tf.py_func(_collect_grads, [flat_grad, stat], tf.float32)
         avg_flat_grad.set_shape(flat_grad.shape)
         avg_grads = tf.split(avg_flat_grad, sizes, axis=0)
         avg_grads_and_vars = [(tf.reshape(g, v.shape), v)
                     for g, (_, v) in zip(avg_grads, grads_and_vars)]
         return avg_grads_and_vars
 
-def check_synced(sess, comm, tfstat):
+def check_synced(comm, localval):
     """
     Check that 'tfstat' evaluates to the same thing on every MPI worker
     """
-    localval = sess.run(tfstat)
```

Comment:
this is the offender - note that it is called in check_synced that is called in _collect_grads that is called within the graph

Reply:
So what caused this to stop working?

Reply:
Also the check_synced docstring should probably be updated.   Ideally it would also say exactly what kind of error this is supposed to cache.